pkg/controller/perconaservermongodb/version_test.go

```diff
@@ -307,3 +328,528 @@ func Test_majorUpgradeRequested(t *testing.T) {
+			},
+		},
+		{
+			name: "TestClusterWideHelmDeploy",
+			cr: api.PerconaServerMongoDB{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "some-name",
+				},
+				Spec: api.PerconaServerMongoDBSpec{
+					Image: "percona/percona-server-mongodb:5.0.11-10",
+					Replsets: []*api.ReplsetSpec{
+						{
+							Name: "rs0",
+							Size: 3,
+							VolumeSpec: &api.VolumeSpec{
+								PersistentVolumeClaim: api.PVCSpec{
+									PersistentVolumeClaimSpec: &corev1.PersistentVolumeClaimSpec{
+										Resources: corev1.ResourceRequirements{
+											Requests: map[corev1.ResourceName]resource.Quantity{
+												corev1.ResourceStorage: q,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				Status: api.PerconaServerMongoDBStatus{
+					Size: 4,
+				},
+			},
+			want: VersionMeta{
+				Apply:              "disabled",
+				Version:            version.Version,
+				HelmDeploy:         true,
+				ClusterWideEnabled: true,
+				ClusterSize:        4,
+			},
+			clusterWide: true,
+			helmDeploy:  true,
+		},
+	}
+	currentNs := "test-namespace"
+	size := int32(1)
+	operatorName := "percona-server-mongodb-operator"
+	operatorDepl := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      operatorName,
+			Namespace: currentNs,
+			Labels:    make(map[string]string),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &size,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": "percona-server-mongodb-operator",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"name": "percona-server-mongodb-operator",
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: "percona-server-mongodb-operator",
+					Containers: []corev1.Container{
+						{
+							Name: "percona-server-mongodb-operator",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Setenv(k8s.WatchNamespaceEnvVar, currentNs)
+			if tt.clusterWide {
+				t.Setenv(k8s.WatchNamespaceEnvVar, "")
+			}
+			if tt.helmDeploy {
+				operatorDepl.Labels["helm.sh/chart"] = operatorName
+			} else {
+				delete(operatorDepl.Labels, "helm.sh/chart")
+			}
+
+			scheme := k8sruntime.NewScheme()
+			if err := clientgoscheme.AddToScheme(scheme); err != nil {
+				t.Fatal(err, "failed to add client-go scheme")
+			}
+			if err := apis.AddToScheme(scheme); err != nil {
+				t.Fatal(err, "failed to add apis scheme")
+			}
+
+			cl := fake.NewClientBuilder().WithScheme(scheme).WithObjects(&tt.cr, &operatorDepl).Build()
+			sv := &version.ServerVersion{Platform: version.PlatformKubernetes}
+			r := &ReconcilePerconaServerMongoDB{
+				client:        cl,
+				scheme:        scheme,
+				serverVersion: sv,
+			}
+			log := logf.Log.WithName(tt.name)
+
+			if err := r.setCRVersion(context.TODO(), &tt.cr); err != nil {
+				t.Error(err, "set CR version")
```

Comment:
Is ok to continue the test if this error occurs? Do you maybe want `t.Fatal` here?

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1139/commits/9d0afcac8604583fc066a8829644e63ec3820b12